File: C Programs/Week1/arrayopt.c
#include <stdio.h>
#define MAX 100
void createArray(int arr[], int *size) {
    printf("Enter number of elements: ");
    scanf("%d", size);
    printf("Enter elements:\n");
    for (int i = 0; i < *size; i++) {
        scanf("%d", &arr[i]);
    }
}
void insertElement(int arr[], int *size) {
    int value, pos;
    printf("Enter value to insert: ");
    scanf("%d", &value);
    printf("Enter position to insert (0-indexed): ");
    scanf("%d", &pos);
    if (pos >= 0 && pos <= *size) {
        for (int i = *size; i > pos; i--) {
            arr[i] = arr[i - 1];
        }
        arr[pos] = value;
        (*size)++;
    } else {
        printf("Invalid position!\n");
    }
}
void modifyElement(int arr[], int size) {
    int pos, newValue;
    printf("Enter index to modify: ");
    scanf("%d", &pos);
    if (pos >= 0 && pos < size) {
        printf("Enter new value: ");
        scanf("%d", &newValue);
        arr[pos] = newValue;
    } else {
        printf("Invalid index!\n");
    }
}
void deleteElement(int arr[], int *size) {
    int pos;
    printf("Enter position to delete (0-indexed): ");
    scanf("%d", &pos);
    if (pos >= 0 && pos < *size) {
        for (int i = pos; i < *size - 1; i++) {
            arr[i] = arr[i + 1];
        }
        (*size)--;
    } else {
        printf("Invalid index!\n");
    }
}
void displayArray(int arr[], int size) {
    printf("Current Array: ");
    for (int i = 0; i < size; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
}
int main() {
    int arr[MAX], size = 0, choice;
    createArray(arr, &size);
    while (1) {
        printf("\nMenu:\n1. Insert\n2. Modify\n3. Delete\n4. Display\n5. Exit\nEnter your choice: ");
        scanf("%d", &choice);
        switch (choice) {
            case 1:
                insertElement(arr, &size);
                break;
            case 2:
                modifyElement(arr, size);
                break;
            case 3:
                deleteElement(arr, &size);
                break;
            case 4:
                displayArray(arr, size);
                break;
            case 5:
                return 0;
            default:
                printf("Invalid choice!\n");
        }
    }
    return 0;
}

File: Python Programs/Week1/arrayopt.py
arr = []
def accept(n):
    global arr
    a = []
    print("Enter", n, "elements:")
    for i in range(n):
        e = int(input(f"Element {i+1}: "))
        a.append(e)
    arr = a
def insert():
    pos = int(input("Enter position to insert at (starting from 1): "))
    val = int(input("Enter element to insert: "))
    if 1 <= pos <= len(arr) + 1:
        arr.insert(pos - 1, val)
    else:
        print("Invalid position!")
def delete():
    pos = int(input("Enter position to delete (starting from 1): "))
    if 1 <= pos <= len(arr):
        t = arr.pop(pos - 1)
        print(t, "removed from array")
    else:
        print("Invalid position!")
def modify():
    pos = int(input("Enter position to modify (starting from 1): "))
    if 1 <= pos <= len(arr):
        val = int(input("Enter new value: "))
        arr[pos - 1] = val
    else:
        print("Invalid position!")
def display():
    print("The array: ")
    for i in range(len(arr)):
        print(arr[i], end=" ")
    print()
def main():
    n = int(input("Enter number of elements to accept: "))
    accept(n)
    while True:
        print("\nMenu:")
        print("1. Insert")
        print("2. Delete")
        print("3. Modify")
        print("4. Display")
        print("5. Exit")
        ch = input("Enter your choice: ")
        if ch == '1':
            insert()
        elif ch == '2':
            delete()
        elif ch == '3':
            modify()
        elif ch == '4':
            display()
        elif ch == '5':
            print("Exiting...")
            break
        else:
            print("Invalid choice!")
main()
